plugins/module_utils/network/sonic/config/ospf_area/ospf_area.py

```diff
@@ -1140,15 +1144,41 @@ def build_area_delete_ranges_requests(self, request_root, commands, have):
                 continue
 
             partial_deletes = True
-            if "advertise" in range_c:
-                requests.append({"path": request_root + "/ranges/range=" + range_string + "/config/advertise", "method": "DELETE"})
             if "cost" in range_c:
                 requests.append({"path": request_root + "/ranges/range=" + range_string + "/config/metric", "method": "DELETE"})
             if "substitute" in range_c:
                 # it actually is mispelled as substitue in REST
                 requests.append({"path": request_root + "/ranges/range=" + range_string + "/config/substitue-prefix",
                                  "method": "DELETE"})
+            # advertise cannot be deleted so set to default true
+            if range_c.get("advertise") is False:
+                requests.append({"path": request_root + "/ranges/range=" + range_string + "/config/advertise", "method": "PATCH",
+                                 "data": {"openconfig-ospfv3-ext:advertise": True}})
```

Comment:
PATCH data pointing to ospfv3-ext

Reply:
Also, advertise can be set to `False` if there is no `cost` (Cost and advertise=false are mutually exclusive)
In cli, it is done by `not-advertise`.